src/plugins/condition/criterion/AllTelemetryCriterion.js

```diff
@@ -127,20 +128,37 @@ export default class AllTelemetryCriterion extends TelemetryCriterion {
                 options
             ));
 
+        let telemetryDataCache = {};
         return Promise.all(telemetryRequests)
             .then(telemetryRequestsResults => {
                 let latestDatum;
+                let latestTimestamp;
+                const timeSystems = this.openmct.time.getAllTimeSystems();
+                const timeSystem = this.openmct.time.timeSystem();
+
                 telemetryRequestsResults.forEach((results, index) => {
                     latestDatum = results.length ? results[results.length - 1] : {};
-                    if (index < telemetryRequestsResults.length-1) {
-                        if (latestDatum) {
-                            this.telemetryDataCache[latestDatum.id] = this.computeResult(latestDatum);
-                        }
+                    if (latestDatum) {
```

Comment:
looks like latestDatum will never be undefined. Do we need this if check?

Reply:
good point. removed